lib/ansible/plugins/inventory/kubevirt.py

```diff
@@ -0,0 +1,178 @@
+                environment variable.
+          cert_file:
+              description:
+              - Path to a certificate used to authenticate with the API. Can also be specified via K8S_AUTH_CERT_FILE
+                environment variable.
+          key_file:
+              description:
+              - Path to a key file used to authenticate with the API. Can also be specified via K8S_AUTH_HOST
+                environment variable.
+          ssl_ca_cert:
+              description:
+              - Path to a CA certificate used to authenticate with the API. Can also be specified via
+                K8S_AUTH_SSL_CA_CERT environment variable.
+          verify_ssl:
+              description:
+              - "Whether or not to verify the API server's SSL certificates. Can also be specified via
+                K8S_AUTH_VERIFY_SSL environment variable."
+              type: bool
+          namespaces:
+              description:
+              - List of namespaces. If not specified, will fetch all virtual machines for all namespaces user is authorized
+                to access.
+
+    requirements:
+    - "python >= 2.7"
+    - "openshift >= 0.6"
+    - "PyYAML >= 3.11"
+'''
+
+EXAMPLES = '''
+# File must be named kubevirt.yaml or kubevirt.yml
+
+# Authenticate with token, and return all virtual machines for all namespaces
+plugin: kubevirt
+connections:
+ - host: https://kubevirt.io
+   token: xxxxxxxxxxxxxxxx
+   ssl_verify: false
+
+# Use default config (~/.kube/config) file and active context, and return objects for a specific namespace
+plugin: kubevirt
+connections:
+  - namespaces:
+      - vms
+'''
+
+from ansible.plugins.inventory.k8s import K8sInventoryException, InventoryModule as K8sInventoryModule, format_dynamic_api_exc
+
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError:
+    pass
+
+
+class InventoryModule(K8sInventoryModule):
+    NAME = 'kubevirt'
+
+    transport = 'oc'
+
+    def fetch_objects(self, connections):
+        client = self.get_api_client()
+
+        if connections:
+            for connection in connections:
+                client = self.get_api_client(**connection)
+                name = connection.get('name', self.get_default_host_name(client.configuration.host))
+                if connection.get('namespaces'):
+                    namespaces = connection['namespaces']
+                else:
+                    namespaces = self.get_available_namespaces(client)
+                for namespace in namespaces:
+                    self.get_vms_for_namespace(client, name, namespace)
+        else:
+            name = self.get_default_host_name(client.configuration.host)
+            namespaces = self.get_available_namespaces(client)
+            for namespace in namespaces:
+                self.get_vms_for_namespace(client, name, namespace)
+
+    def get_vms_for_namespace(self, client, name, namespace):
```

Comment:
We use `vms` name whereas we fetch `vmis`. I suggest to be consistent. In addition why we are fetching vmis and not vms?

Reply:
Because VMIs has IPs, VMs don't so it doesn't make much sense to fetch them, but if you think we should, we can add it.